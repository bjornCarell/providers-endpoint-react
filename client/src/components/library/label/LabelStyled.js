import styled from 'styled-components';

export const Label = styled.label`
  display: ${props => (props.block ? 'block' : 'inline')};
  font-size: ${props =>
    props.medium
      ? ({ theme }) => theme.fontSize.medium
      : ({ theme }) => theme.fontSize.small};
  font-weight: ${props =>
    props.medium
      ? ({ theme }) => theme.fontWeight.large
      : ({ theme }) => theme.fontWeight.medium};
  margin-bottom: ${props =>
    props.block ? ({ theme }) => theme.spacing.xss : 0};
  margin-top: ${props =>
    props.block ? ({ theme }) => theme.spacing.medium : 0};
`;
