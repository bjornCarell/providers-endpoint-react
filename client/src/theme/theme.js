export const theme = {
  color: {
    primary: '#004146',
    primaryLight: '#427783',
    primaryBright: '#E8F9F8',
    secondary: '#F89572',
    secondaryLight: '#FDDFD4',
    secondaryDark: '#FF8A61',
    secondaryBright: '#FFE8E5',
    white: '#FFFFFF',
    light: '#fcfcfc',
    black: '#262626',
    grey: '#F5F6F7'
  },
  spacing: {
    xss: '0.5rem',
    xs: '0.75rem',
    small: '1rem',
    medium: '2rem',
    large: '4rem',
    xl: '8rem',
    xxl: '16rem'
  },
  fontSize: {
    small: '1.2rem',
    medium: '1.6rem',
    large: '2.6rem',
    xl: '4.22rem',
    xxl: '6.85rem'
  },
  fontWeight: {
    small: 200,
    medium: 400,
    large: 600,
    xl: 800
  },
  animation: {
    shadowAround: '0px 0px 10px 0px rgba(0,0,0,0.1)',
    shadowAroundHover: '0px 0px 16px 0px rgba(0,0,0,0.13)',
    shadowBottom: '0px 4px 8px 0 rgba(0,0,0,0.1)',
    shadowBottomHover: '0px 8px 8px 0 rgba(0,0,0,0.12)',
    radiusS: '0.375rem',
    radiusM: '0.875rem',
    radiusL: '2rem',
    transition: 'all .25s cubic-bezier(0.4,0,0.2,1)'
  },
  zIndex: {
    sidebar: 4
  }
};
