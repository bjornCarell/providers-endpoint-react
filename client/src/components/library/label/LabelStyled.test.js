import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import { theme } from '../../../theme/theme';
import 'jest-styled-components';
import { Label } from './LabelStyled';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<Label theme={theme}>Providers</Label>, div);
});

it('renders with default conditional styles', () => {
  const tree = renderer.create(<Label theme={theme}>Providers</Label>).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', `${theme.fontSize.small}`);
  expect(tree).toHaveStyleRule('font-weight', `${theme.fontWeight.medium}`);
  expect(tree).toHaveStyleRule('margin-bottom', '0');
  expect(tree).toHaveStyleRule('margin-top', '0');
});

it('renders with default conditional styles', () => {
  const tree = renderer
    .create(
      <Label theme={theme} medium block>
        Providers
      </Label>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', `${theme.fontSize.medium}`);
  expect(tree).toHaveStyleRule('font-weight', `${theme.fontWeight.large}`);
  expect(tree).toHaveStyleRule('margin-bottom', `${theme.spacing.xss}`);
  expect(tree).toHaveStyleRule('margin-top', `${theme.spacing.medium}`);
});
