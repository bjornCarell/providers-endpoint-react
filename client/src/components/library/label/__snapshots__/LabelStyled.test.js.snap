// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with default conditional styles 1`] = `
.c0 {
  display: inline;
  font-size: 1.2rem;
  font-weight: 400;
  margin-bottom: 0;
  margin-top: 0;
}

<label
  className="c0"
>
  Providers
</label>
`;

exports[`renders with default conditional styles 2`] = `
.c0 {
  display: block;
  font-size: 1.6rem;
  font-weight: 600;
  margin-bottom: 0.5rem;
  margin-top: 2rem;
}

<label
  className="c0"
>
  Providers
</label>
`;
